src/CMakeLists.txt

```diff
@@ -0,0 +1,184 @@
+if ( ${T8CODE_BUILD_AS_SHARED_LIBRARY} )
+    add_library( T8 SHARED )
+    set_target_properties( T8 PROPERTIES POSITION_INDEPENDENT_CODE ON )
+else()
+    add_library( T8 STATIC )
+endif()
+
+set_target_properties( T8 PROPERTIES OUTPUT_NAME t8 )
+
+target_compile_definitions( T8 PUBLIC T8_CMAKE_BUILD )
+target_compile_definitions( T8 PUBLIC T8_CC="${CMAKE_C_COMPILER}" )
+target_compile_definitions( T8 PUBLIC T8_CFLAGS="${CMAKE_C_FLAGS}" )
+target_compile_definitions( T8 PUBLIC T8_CPP="${CMAKE_CXX_COMPILER}" )
+target_compile_definitions( T8 PUBLIC T8_CPPFLAGS="${CMAKE_CXX_FLAGS}" )
+target_compile_definitions( T8 PUBLIC T8_LDFLAGS="${CMAKE_SHARED_LINKER_FLAGS}" )
+target_compile_definitions( T8 PUBLIC T8_LIBS="???" ) # Not sure what this flag is supposed to be...
```

Comment:
This flag is only used en `t8.c` line 158. Removing it breaks the build:
`[ 42%] Building C object src/CMakeFiles/T8.dir/t8.c.o
/Users/adutka/t8code/src/t8.c:158:50: error: use of undeclared identifier 'T8_LIBS'
  t8_global_productionf ("%-*s %s\n", w, "LIBS", T8_LIBS);`
 I don't quite undersand what was the purpose of this line, but I find it quite useless (at least in the context of a CMake build)

Reply:
It is defined here: https://github.com/DLR-AMR/t8code/blob/bd82ee6ee4e87a610abd8f76a45a3c0cfbbeb422/Makefile.t8.pre.in#L27

Can we reproduce this with cmake?

Reply:
Looks like this would not be trivial and a little bit hacky (see https://stackoverflow.com/questions/34165365/retrieve-all-link-flags-in-cmake for example)

Reply:
Closing this thread. Feel free to reopen if you disagree with me on this matter.